tools/ui-components/src/form-group/form-group.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useMemo } from 'react';
+import { FormContext } from '../form-context';
+import { FormGroupProps } from './types';
+
+let variantClass = '';
+const hasSuccess =
+  "[&_span:has[data-testid='help-block']]: text-foreground-info [&_label]: text-foreground-info";
+const hasWarning =
+  "[&_span:has[data-testid='help-block']]: text-foreground-warning [&_label]: text-foreground-warning";
+const hasError =
+  "[&_span:has[data-testid='help-block']]: text-foreground-danger [&_label]: text-foreground-danger";
```

Comment:
Since validation is being passed down using context, styles should be added in their respective component if needed by checking for the validation state. 

it is not a good idea to use `data-testid` as a selector since it is used merely for testing.

Reply:
I need this PR, or ControlLabel PR to be merged, to finish the validation styles. 

True the selector is bad, we don't have ControlLabel in yet, so I can't add style for the `span` in the ControlLabel, this is a placeholder until one is pushed, so I don't do the research again.

Reply:
deleted help block styles, because I found component that have them, and it isn't used in formGroup in anywhere, and if it's used we can add quick fix their instead

for context the HelpBlock https://github.com/Sboonny/freeCodeCamp/blob/9959cd89ed439aac3406c87b78dc04c985d9582e/tools/ui-components/src/help-block/help-block.tsx#L1-L17